### PR TITLE
add Url VO & UrlValidator

### PR DESCRIPTION
**Pull request by angyvolin**

### CODE REVIEW

**Review by 2DSharp — CHANGES_REQUESTED** [review]
The test fails on using non-http based urls, for example `ftp://`. Needs fixing.

**Comment by 2DSharp** [comment]
Also, is there a specific reason for not using `filter_var` for validating the URL?

**Comment by angyvolin** [comment]
> Also, is there a specific reason for not using `filter_var` for validating the URL?

https://d-mueller.de/blog/why-url-validation-with-filter_var-might-not-be-a-good-idea/

**Comment by 2DSharp** [comment]
You seem to have copied the code from `Symfony/UrlValidator` (https://github.com/symfony/Validator/blob/master/Constraints/UrlValidator.php). Symfony operates under the MIT license. Please add the license for this, here are some suggestions: https://softwareengineering.stackexchange.com/questions/297217/using-code-under-apache-2-0-license-in-my-program-using-mit-license

**Review by 2DSharp — APPROVED** [review]

File: tests/UnitTests/Validator/UrlValidatorTest.php [hunk 1]
```diff
@@ -91,6 +91,18 @@ public function correctUrl()
...
+            ],
+            yield [
+                'ws://192.168.1.15:8000',
+            ],
+            yield [
+                'wss://192.168.1.15:8000/messages',
+            ],
             yield [
                 '//yahoo.com',
             ],
```
Comment (2DSharp) [inline tests/UnitTests/Validator/UrlValidatorTest.php:108]: Please add some more tests on invalid values from https://d-mueller.de/blog/why-url-validation-with-filter_var-might-not-be-a-good-idea/

**Review by 2DSharp — CHANGES_REQUESTED** [review]

**Pull request closed without merging**